### PR TITLE
Assignment3

### PR DESCRIPTION
**Pull request by FenrisStargazer**
abrand3 / FenrisStargazer, no partner

### CODE REVIEW

**Comment by Clockknight** [comment]
I will be peer reviewing this.

**Comment by Wvvern** [comment]
Will review

**Review by Wvvern — COMMENTED** [review]
### Assignment
- Create *two* interfaces. One to represent the `JokeService` and one interface to represent outputting the joke to the screen (`Console.WriteLine`). ✔
- The `JokeService` will need to have the interface applied to it. ✔
- Create an implementation of the output interface that writes out the joke to the console. ✔
- Implement the `Jester` class. It should take in both interfaces as dependencies. These dependencies should be null checked. ✔
- The `Jester` class `TellJoke()` method should retrieve a joke from the `JokeService`. If the joke contains "Chuck Norris", skip it and get another. The joke should be written to the output dependency. ✔
- Unit test the Jester class. Code coverage should be above 90% for this class. Moq will make this significantly easier. ✔


- Be sure you enable:
  - Nullability for all projects  ✔
  - Set `LangVersion` and the `TargetFramework` to the latest released versions available (preview versions optional) ✔
  - Ensure that you turn on code analysis for all projects(EnableNETAnalyzers) ✔
- For this assignment, always use `Assert.AreEqual<T>()` (the generic version)  ❌✔ (N/A)
- **Ensure there are no errors or warnings (including code analysis warnings)** ✔? ([IDE1006](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/naming-rules))
- **All of the above should be unit tested** ❌
- **Choose simplicity over complexity** ✔

### Extra Credit
- Unit test your implementation that writes the joke out to the screen. How hard could it be to unit test a single line method ;)? ❌
- The Geek jokes API that is being used can also return jokes in a JSON format. Update the `JokeService` to retrieve jokes using JSON. `GetJoke` should still return a string. ✔

File: CanHazFunny/CanHazFunny/JokeService.cs [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
-    public class JokeService
+    public class JokeService : IJokeService
     {
         private HttpClient HttpClient { get; } = new();
 
         public string GetJoke()
         {
-            string joke = HttpClient.GetStringAsync("https://geek-jokes.sameerkumar.website/api").Result;
+            //This most assuredly isn't the proper or good way to do it but idk it kinda works and I can't get the 
+            //stupid deserialization nonsense to work with the time left. So voila, jury-rigged solution.
```
Comment (mmwoodfo) [inline CanHazFunny/CanHazFunny/JokeService.cs:14]: Yeah, this technically isn't deserializing the JSON into an object, which is what the extra credit is asking for. This is just trimming the string as you mentioned. 

Check out the assignment solution or this PR for the answer:
https://github.com/IntelliTect-Samples/EWU-CSCD371-2023-Winter/pull/74

File: CanHazFunny/CanHazFunny.Tests/JesterTests.cs [hunk 2]
```diff
@@ -1,10 +1,38 @@
...
+        }
+
+        [TestMethod]
+        public void TellJoke_PrintsJoke()
+        {
+            //Suggestion from peer PR review
+            Mock<IJokeService> MockJokeService = new();
+            Mock<IPrintJoke> MockJokePrinter = new();
+            Jester jester = new(MockJokePrinter.Object, MockJokeService.Object);
+            MockJokeService.Setup(x => x.GetJoke()).Returns("incredible joke here");
```
Comment (mmwoodfo) [inline CanHazFunny/CanHazFunny.Tests/JesterTests.cs:31]: Nice job!

**Pull request closed without merging**